src/main/scala/ai/privado/passes/PropertyCollectorPass.scala

```diff
@@ -0,0 +1,391 @@
+  }
+
+  // Used to extract (name, value) pairs from a bean config file
+  private def XMLParserBean(xmlPath: String, builder: DiffGraphBuilder): List[(String, String)] = {
+    try {
+      val xml = XML.loadFile(xmlPath)
+      val nameValuePairs = (xml \\ "bean").flatMap { bean =>
+        {
+          var result: (String, String) = ("", "")
+          val className: String        = bean \@ "class"
+          (bean \\ "property").map { prop =>
+            {
+              // Search for property tags inside a bean
+              val propValue = prop \@ "value"
+              if (propValue.startsWith("$") && propValue.endsWith("}")) {
+                val value = resolvePlaceholderValuesXML(
+                  propValue.substring(2, propValue.length - 1)
+                ) // Pass placeholder name without ${ and }
+                if (value.nonEmpty) {
+                  result = ((prop \@ "name"), value)
+                }
+              } else {
+                result = ((prop \@ "name"), propValue)
+              }
+
+            }
+
+            val members = getMember((prop \@ "name"), className);
+            if (members.nonEmpty) {
+              val propertyNode = NewJavaProperty().name(result._1).value(result._2)
+              val member       = members.head
+              if (member != null) {
+                builder.addEdge(propertyNode, member, EdgeTypes.IS_USED_AT)
+                builder.addEdge(member, propertyNode, EdgeTypes.ORIGINAL_PROPERTY);
+              }
+            }
+            result
+          }
+        }
+      }
+
+      return nameValuePairs.toList
+        .collect { case (name, value) => if (value.nonEmpty) (name, value) else ("", "") }
+        .filter { case (name, value) =>
+          name.nonEmpty && value.nonEmpty // Filter out name, value pairs which could not be resolved
+        }
+    } catch {
+      case e: Throwable => println(e)
+    }
+
+    List[("", "")]()
+
+  }
+
+  private def loadAndConvertXMLtoProperties(file: String, builder: DiffGraphBuilder): List[(String, String)] = {
+    val properties  = new Properties();
+    val inputStream = better.files.File(file).newInputStream
+    try {
+      properties.loadFromXML(inputStream)
+      properties.propertyNames.asScala.toList
+        .collect(p => (p.toString, properties.getProperty(p.toString)))
+    } catch {
+      case _: Throwable => {
+        XMLParserBean(file, builder)
+      }
+    }
+  }
+
+  private def resolvePlaceholderValuesXML(placeholder: String): String = {
+    val propertyFiles: List[String] = configFiles(projectRoot, Set(".properties"))
+    propertyFiles.foreach(file => {
+      // Search across properties to find the required
+      loadFromProperties(file).foreach(propertyValue => {
+        val (name, value) = propertyValue;
+        if (name.equals(placeholder)) return value;
+      })
+    })
+    ""
+  }
+
+  private def matchingLiteralsInGetPropertyCalls(propertyName: String): List[Literal] = cpg.literal
```

Comment:
I guess these kind of linking functions are not needed in this file, remove all of them